### PR TITLE
downsampling original img / upsampling disparity img

### PR DESCRIPTION
**Pull request by hokwangchoi**
issue #613

### CODE REVIEW

File: stereo_image_proc/src/nodelets/disparity.cpp [hunk 1]
```diff
@@ -196,8 +242,32 @@ void DisparityNodelet::imageCb(const ImageConstPtr& l_image_msg,
   const cv::Mat_<uint8_t> l_image = cv_bridge::toCvShare(l_image_msg, sensor_msgs::image_encodings::MONO8)->image;
   const cv::Mat_<uint8_t> r_image = cv_bridge::toCvShare(r_image_msg, sensor_msgs::image_encodings::MONO8)->image;
 
+  // Downsampling
+  int downsampling_factor;
+  getPrivateNodeHandle().param("downsampling_factor", downsampling_factor, 1);
```
Comment (JWhitleyWork) [inline stereo_image_proc/src/nodelets/disparity.cpp]: Getting this parameter should be handled up in `onInit` as with other parameters.

File: stereo_image_proc/src/nodes/stereo_image_proc.cpp [hunk 2]
```diff
@@ -106,6 +106,10 @@ int main(int argc, char **argv)
   if (private_nh.getParam("queue_size", queue_size))
     shared_params["queue_size"] = queue_size;
 
+  int downsampling_factor;
+  if (private_nh.getParam("downsampling_factor", downsampling_factor))
+    shared_params["downsampling_factor"] = downsampling_factor;
```
Comment (JWhitleyWork) [inline stereo_image_proc/src/nodes/stereo_image_proc.cpp:111]: Could you please add this to `stereo_image_proc/launch/stereo_image_proc.launch` with the default value of 1?

**Review by JWhitleyWork — CHANGES_REQUESTED** [review]

File: stereo_image_proc/src/nodelets/disparity.cpp (hunk 1 above)
Comment (hokwangchoi) [reply stereo_image_proc/src/nodelets/disparity.cpp]: done

File: stereo_image_proc/src/nodes/stereo_image_proc.cpp (hunk 2 above)
Comment (hokwangchoi) [reply stereo_image_proc/src/nodes/stereo_image_proc.cpp:111]: Yes, done

**Comment by JWhitleyWork** [comment]
@choi0330 Would you mind forward-porting this to the `noetic` branch?

**Comment by hokwangchoi** [comment]
> @choi0330 Would you mind forward-porting this to the `noetic` branch?

Done. #623

**Comment by JWhitleyWork** [comment]
Thank you!

**Pull request merged**